views/application.py

```diff
@@ -93,3 +93,28 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+    
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+        view details of application which should not be exposed directly 
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self,request,*args,**kwargs):
+        application = Application.objects.get(pk=kwargs['id'])
```

Comment:
Instead of looking up using `[]`, use [`.get(name, value)`](https://www.w3schools.com/python/ref_dictionary_get.asp)